### ISSUE TITLE
Crash after switching from ES6 -> ES5 Mode

### ISSUE THREAD

**Issue by mousetraps** [Microsoft/nodejstools#1016]
Still working on repro steps:
```
---------------------------
Assertion Failed: Abort=Quit, Retry=Debug, Ignore=Continue
---------------------------
An unhandled exception is about to terminate the process:

No such interface supported (Exception from HRESULT: 0x80004002 (E_NOINTERFACE))

System.InvalidCastException: No such interface supported (Exception from HRESULT: 0x80004002 (E_NOINTERFACE))

   at EnvDTE._DTE.get_Properties(String Category, String Page)

   at Microsoft.NodejsTools.SmartIndent.GetDesiredIndentation(ITextSnapshotLine line) in C:\src\nodejstools\Nodejs\Product\Nodejs\SmartIndent.cs:line 51

   at Microsoft.VisualStudio.Text.Editor.Implementation.SmartIndentationService.GetDesiredIndentation(ITextView textView, ITextSnapshotLine line)

   at Microsoft.VisualStudio.Text.Editor.Implementation.CaretElement.MapXCoordinate(ITextViewLine textLine, Double xCoordinate, Boolean userSpecifiedXCoordinate)

   at Microsoft.VisualStudio.Text.Editor.Implementation.CaretElement.MoveTo(ITextViewLine textLine, Double xCoordinate, Boolean captureHorizontalPosition)

   at Microsoft.VisualStudio.Text.Editor.Implementation.CaretElement.MoveTo(ITextView......

<truncated>
---------------------------
Abort   Retry   Ignore   
---------------------------
```

**Comment by mjbvz** [comment]
Is this a dup of #1014? This also involved the smart indenter and a similar call stack if I recall, but I was seeing it in the interactive window instead.

**Comment by mousetraps** [comment]
Not sure - this brought down all of VS.

**Comment by mjbvz** [comment]
The crash location and local call stack are the same as #1014, so it's likely the exception is just not being handled in this case. I submitted a fix with the previous one with #1015.

**Comment by mousetraps** [comment]
Okay, let's try and get that fix in ASAP then because it's almost impossible to test the editor without it.

**Comment by mjbvz** [comment]
@mousetraps addressed comments on #1015 and merged in. Switching IntelliSense modes is working for me but please let me know if you are still seeing the issue.
